MyPerfectOnboardingApplication/Tests/MyPerfectOnboarding.App.Tests/Api/Controllers/ListControllerApplicationTests.cs

```diff
@@ -0,0 +1,282 @@
+            container.Register(_guidGenerator);
+            _listRepository = Substitute.For<IListRepository>();
+            container.Register(_listRepository);
+        }
+
+        [Test]
+        public async Task GetAllItems_ThreeTimes_DatabaseOnlyReadOnce()
+        {
+            _listRepository.GetAllItemsAsync().Returns(_items);
+
+            var message1 = await _listController.ExecuteAction(controller => controller.GetAsync());
+            message1.TryGetContentValue(out IEnumerable<IListItem> resultItems1);
+
+            var message2 = await _listController.ExecuteAction(controller => controller.GetAsync());
+            message2.TryGetContentValue(out IEnumerable<IListItem> resultItems2);
+
+            var message3 = await _listController.ExecuteAction(controller => controller.GetAsync());
+            message3.TryGetContentValue(out IEnumerable<IListItem> resultItems3);
+
+            Assert.Multiple(() => {
+                Assert.That(message1.StatusCode, Is.EqualTo(HttpStatusCode.OK));
+                Assert.That(message2.StatusCode, Is.EqualTo(HttpStatusCode.OK));
+                Assert.That(message3.StatusCode, Is.EqualTo(HttpStatusCode.OK));
+
+                Assert.That(resultItems1, Is.EqualTo(_items).UsingListItemComparer());
+                Assert.That(resultItems2, Is.EqualTo(_items).UsingListItemComparer());
+                Assert.That(resultItems3, Is.EqualTo(_items).UsingListItemComparer());
+
+                _listRepository.Received(1).GetAllItemsAsync();
+            });
+        }
+
+        [Test]
+        public async Task DeleteItem_SameItemTwice_DeleteItemAndSecondTimeReturnNotFound()
+        {
+            var itemIdToDelete = _items[1].Id;
+            _listRepository.GetAllItemsAsync().Returns(_items);
+
+            var message1 = await _listController.ExecuteAction(controller => controller.DeleteAsync(itemIdToDelete));
+            var message2 = await _listController.ExecuteAction(controller => controller.DeleteAsync(itemIdToDelete));
+
+            Assert.Multiple(() => {
+                Assert.That(message1.StatusCode, Is.EqualTo(HttpStatusCode.NoContent));
+                Assert.That(message2.StatusCode, Is.EqualTo(HttpStatusCode.NotFound));
+
+                _listRepository.Received(1).GetAllItemsAsync();
+                _listRepository.Received(1).DeleteItemAsync(Arg.Is<Guid>(id => id.Equals(itemIdToDelete)));
+            });
+        }
+
+        [Test]
+        public async Task AddItemAndGetAllItems_Item_ReturnsAllItemsIncludedNewOne()
+        {
+            _listRepository.GetAllItemsAsync().Returns(_items);
+
+            var stringId = "22AC59B7-9517-4EDD-9DDD-EB418A7C1678";
+            var id = Guid.Parse(stringId);
+            var expectedUri = new Uri($"http://www.aaa.com/{stringId}");
+            _location.GetListItemLocation(id).Returns(expectedUri);
+
+            var stringTime = "2150-12-05";
+            var addedItemCreationTime = DateTime.Parse(stringTime);
```

Comment:
You, kind of, didn't address the main issue... Why don't you use `addedItem.LastUpdateTime` and shorten the setup as suggested in https://github.com/HanaNavratilova/kentico-onboarding-cs/pull/4#discussion_r262051980?